apps/guardian-ui/src/setup/SetupContext.tsx

```diff
@@ -229,6 +220,13 @@ export const SetupContextProvider: React.FC<SetupContextProviderProps> = ({
         configGenParams: state.configGenParams,
       })
     );
+
+    return () => {
+      // Clear local storage on setup complete.
+      // This happens when we transition to admin experience.
+      state.progress === SetupProgress.SetupComplete &&
+        localStorage.removeItem(LOCAL_STORAGE_SETUP_KEY);
+    };
```

Comment:
Do we want this in the useEffect teardown function? Seems to me like the useEffect should be an `if / else`:

```ts
if (state.progress === SetupProgress.SetupComplete) {
  localStorage.removeItem(...)
} else {
  localStorage.setItem(...)
}
```

Reply:
To clarify, the way this is written now would cause the `removeItem` to run, but then immediately the useEffect would run with the new dependencies and re-write to `localStorage` again.

Reply:
> Do we want this in the useEffect teardown function? Seems to me like the useEffect should be an if / else

An `if` is sufficient. We only nead to clean up the local storage in the singular case when we got to `SetupComplete`, and are now tearing down `SetupContext`. 

> but then immediately the useEffect would run with the new dependencies and re-write to localStorage again.

Logically, we always (re)boot up to admin experience from this point onwards, thus we are _terminally tearing down `SetupContext`_ and this effect that watches state changes never runs again

Rewritten as an `if` instead of the `&&` shortcut

Reply:
Gotcha, I was missing that implicit relationship. Thanks for the rewrite though, that at least makes sure it behaves correctly, even if the context doesn't get torn down like it should.